src/extension.ts

```diff
@@ -163,10 +176,26 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
       if (e.affectsConfiguration('workbench.colorTheme')) {
         const userSettings = getUserSettings();
         const currentTheme = userSettings.get('workbench.colorTheme', '');
-        const oneMoreTimeThemeList: string[] = <string[]>userSettings.get(SettingsKeys.PreventReloadThemeList, MATERIAL_LIST);
 
+        const oneMoreTimeThemeList: string[] = <string[]>userSettings.get(SettingsKeys.PreventReloadThemeList, MATERIAL_LIST);
         context.globalState.update(LAST_THEME_NEEDS_TO_PERSIST, oneMoreTimeThemeList.findIndex(mat => mat === currentTheme) !== -1);
         themeList = await getThemeList(getExtensionConfig(), getUserSettings());
+
+        if (!themeList.includes(currentTheme)) {
```

Comment:
Are you sure that here it doesn't trigger when the current theme is changed by the extension? Like if someone changes by using the command prompt or the hotkey?

Reply:
I have thought it could be useful for the user when setting a theme to know if it's not in the list and providing a button in order to add it.
Moreover i don't know any vscode event that informs the extensions about new themes/extensions that are installed.
(maybe looking at the extension folder or setting an interval could do the trick, but i preferred this approach)

I could turn the `await` into a `thenable` in order to speed the things up.

Reply:
I fixed it, then i discovered another bug :
getThemeList() returns a themeList without the current theme, so the above condition
``` Typescript
if (!themeList.includes(currentTheme)) {
```
get always true and the popup always appears.

A good side effect of having fixed it is that now the swap is more accurate (because previously it will skip only the first "current theme" while considering the next "current theme", if you take a closer look to the next commit you can see the hidden issue that got solved) 
Fixed it i discovered another little bug:
themeList used by the `changeTheme` could resulting being a little behind the actual themeList settings (i.e. if the user adds or removes themes from the list he has to reload VSC in order to get the new themeList configuration ready, check the _themeList in the next commit to spot the issue)

Reply:
this could be bad for consistency. For now it's good as it is.

Reply:
Not sure what are you referring to:

1) I fixed the issue with the themeList That doesn't contain the currentTheme 
2) the fix automatically improve the random theme swap accuracy
3) i noticed that the themeList used in theme swap could potentially not being aligned with the other functions list

I guess the 3, because the first two could be more difficult to revert 😅

Anyway, i have learned how to request a review, so in the next commit i'll know what to do.
Thanks 😊✌🏼